pkgs/_analyzer_macros/lib/macro_implementation.dart

```diff
@@ -64,13 +67,22 @@ class AnalyzerMacroRunner implements injected.MacroRunner {
   AnalyzerMacroRunner(this._impl);
 
   @override
-  injected.RunningMacro run(Uri uri, String name) => AnalyzerRunningMacro.run(
-    _impl,
-    QualifiedName(uri: '$uri', name: name),
-    _impl._host.lookupMacroImplementation(
+  injected.RunningMacro run(Uri uri, String name) {
+    // TODO(davidmorgan): do this in the analyzer.
+    name = _removePrefix(name);
+    return AnalyzerRunningMacro.run(
+      _impl,
       QualifiedName(uri: '$uri', name: name),
-    )!,
-  );
+      _impl._host.lookupMacroImplementation(
+        QualifiedName(uri: '$uri', name: name),
+      )!,
+    );
+  }
+}
+
+String _removePrefix(String name) {
+  final index = name.indexOf('.');
```

Comment:
Done.

It's import prefixes, so `@prefix0.BuiltValueBuilder()` in the macro output was resulting in name `prefix0.BuiltValueBuilder` here. The prefix is meaningless since we have the URI and what we care about is the name in that URI.